validation.go

```diff
@@ -17,7 +17,7 @@ var log = logging.Logger("routing/record")
 
 // ValidatorFunc is a function that is called to validate a given
 // type of DHTRecord.
-type ValidatorFunc func(string, []byte) error
+type ValidatorFunc func(*pb.Record) error
```

Comment:
As this is a public API, I'd prefer to define a new Record struct:

```go
type Record struct {
  Key string
  Value []byte
  Author peer.ID // If and only if signed. Otherwise, just ignore the author (can't trust it).
}
```

It also prevents the validator from looking at anything it shouldn't.